### ISSUE TITLE
Feature Suggestion -- syntax highlighter

### ISSUE THREAD

**Issue by canfine** [vdm-io/Joomla-Component-Builder#305]
### This is a feature suggestion. 
It could be very usefull feature to make new fieldtype  with syntax highlight  and implement it to the jcb custom scripting textarea
### Expected result
https://highlightjs.org/
or some thing like that [joomla](https://extensions.joomla.org/extension/cedhighlightjs/)
### Actual result
just text area

### Additional comments
jcb takes some of the ide's features to simplify component development. On the other hand, the lack of syntax highlighting complicates the work with it

**Comment by ro-ot** [comment]
This is a wonderful idea, and something we have also been talking about. We are also thinking about code hinting.

How much can you contribute to make this happen?

**Comment by canfine** [comment]
@ro-ot, Sorry, I'm not a native speaker and I'm not an active github user. So I'm not sure I got you right. Explain in detail what is required of me?

**Comment by ro-ot** [comment]
Ok, well I am simply asking, if you can help code this.

**Comment by canfine** [comment]
Ok. I think that this task consists of three parts: 
1) input template 
2) embedding in the copy of the textarea field 
3) replacing the corresponding field types
the first part I will provide

**Comment by canfine** [comment]
Textarea doesn't fit. 
need fieldtype "editor" with a non-default editor,
create an options of  creating the field . In this option user shoud  enter the installed editor like codemirror. and choose theme of highlight.

**Comment by canfine** [comment]
the easiest way to use instead of textarea field type editor field type and
[plagin](https://extensions.joomla.org/extensions/extension/authoring-a-content/admin-interface/switch-editor/)

**Comment by canfine** [comment]
the plugin is not even needed. I
![11](https://user-images.githubusercontent.com/5087049/41765610-4042330a-760d-11e8-8dc9-182e50e6fbb1.PNG)
n the field options with the type "editor" there is an editor item

**Comment by canfine** [comment]
that's all

**Comment by Llewellynvdm** [comment]
The reason we did not go that direction is because the editors post via the site safety filters of Joomla and PHP is not allowed in most cases.

So what happens is, if you save a view/item that has php script in a field, or even JavaScript, the editor breaks it and often strips most of it away.

Have you been able to get it to work with PHP code snippets? I must admit I am speaking about four years ago, so it may have improved since. Just been to occupied to look, can you test this for us and give me some feedback? We are willing to change this, I mean why not... if it will work with PHP and JavaScript without stripping any code.

**Comment by Llewellynvdm** [comment]
Okay I just ran a test with a field in the admin view, and since the field is set as required it keeps saying that the field is empty. I then realized that it is due to the values being kept in the input field build by the editor, and so the form does not want to submit.

Any ideas how I can fix this? My JavaScript is not that great.

**Comment by canfine** [comment]
in com_templates " PHP " saving and highlighting works well. The keyword responsible for the correct display of " php " is " syntax" use search through folder

**Comment by canfine** [comment]
And don't forget that some fields in JCB work with out "<?php ?>". Plane changing their  fieldtype would not work.

**Comment by Llewellynvdm** [comment]
Please test the converted fields that are now editor fields inside the admin view of JCB ([staging branch](https://github.com/vdm-io/Joomla-Component-Builder/archive/staging.zip))

**Comment by canfine** [comment]
it's hard to test for me because I haven't used these fields until now
And now I meet with a mistakes, because I do not understand how they work yet.
> PHP publish Method before
> Target (array) $pks is an array of record primary keys and $value is the published state. 
it will take some time to master these fields. Now I just can say  that highlighting is worked.

**Comment by canfine** [comment]
Same fields in admin view were missed. 1,2) javascript footer and footers 3) Add Custom Import

**Comment by Llewellynvdm** [comment]
Yes the idea is just to test that the field work, in submission/save of the view. Not in relation to its implementation to a component. Since that has not changed, does still work well.

The fields we have skipped are more complex and may take some time to convert. Specially all those related to the custom import option.

I will now start moving other areas to this implementation. I have learned to code without the editorial help of an editor, but this change is very welcome, and will make it much easier for others.

**Comment by ro-ot** [comment]
I have been running some tests and it seems like the editor is not always loading correctly, specially with areas that already has code.
The code does not show until you click in the box. So here I have a field with code, but it does not show.
![image](https://user-images.githubusercontent.com/2594383/41950247-1bd910de-79bd-11e8-9b82-2003d2a2edef.png)
So when I click in the editor it loads the code, but behind the line numbers.
![image](https://user-images.githubusercontent.com/2594383/41950264-34cd53f2-79bd-11e8-8d9a-9a6df1c80646.png)

This is with Joomla 3.8.10

**Comment by canfine** [comment]
Сan you describe what is unstable? And give me a link to download and test version with codemirror, please?

**Comment by Llewellynvdm** [comment]
Well just download the latest JCB (staging branch) and try to save any code area that uses codemirror (Joomla default code editor) and you will see with the new Joomla 3.8.10 it does not work. I have been trying all kinds of validation, filters and yet it still continues to block the submission of the new custom code all over JCB.

I have tested this in other areas of Joomla, and it seems to work. But I can not find the problem ,already spend three days on this, so I am a little fed up.
